src/containers/App/App.js

```diff
@@ -81,10 +81,13 @@ import {
   isReadOnly,
   isWebSocketConnected
 } from '../../reducers';
-import messages from '../../nls/messages_en.json';
+import enMessages from '../../nls/messages_en.json';
+import jaMessages from '../../nls/messages_ja.json';
```

Comment:
We should only load the one file required for the user's language, as we would end up loading several thousand lines of JSON otherwise when we add more languages.

As I mentioned in the issue we should add any language with completed translations to `config.locales.supported` and use that to determine which files can be loaded, or fallback to English as a default if the user's language is not yet supported.
https://github.com/tektoncd/dashboard/blob/aeaee0be5a1bbadcc59d98151d968f2fe3698a01/config_frontend/config.json#L6

The existing code already handles using English as a fallback for any untranslated strings in the selected language so there's no need to include `message_en.json` when loading another language.

Reply:
Thank you @AlanGreene and all, for kind advice.
We are wondering how we can load a specific language file such as messages_ja.json dynamically.
`require` requires a literal and will be specified at compile time, `fs` cannot load modules.
Should you have any suggestion, please let us know.

Reply:
For dynamically loading additional content in the client you can use a [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports).

We're already using it in the [Extension component](https://github.com/tektoncd/dashboard/blob/7c74fcc2161a6117d615cdc7bf7bb0c356718140/src/containers/Extension/Extension.js#L29) for example.

Reply:
Thank you for your advice.

Since it will take some time to deal with the modification of `App.js`, we would like to  remove the change in `App.js` and just let you merge the changes to message_ja.json. 

And because it's also needed to squash the commits, I'm going to close this PR and create a new one afterwards. Or if there is better way to do with that, please let me know.

Reply:
Opening a new PR will be fine 👍 Thank you